vllm/device_allocator/cumem.py

```diff
@@ -0,0 +1,244 @@
+# cumem-based pytorch pluggable allocator to implement sleep mode.
+# other approaches tried but failed:
+# - cuda-python package binding
+# - custom libcuda driver ctypes wrapper
+# both of them failed because of cuda context mismatch.
+# not sure why, they are created from a different context.
+# the only successful approach is to call cuda driver API in C.
+import dataclasses
+from contextlib import contextmanager
+from typing import Callable, Dict, Optional, Tuple, Union
+
+import torch
+
+from vllm.utils import is_pin_memory_available
+
+
+def find_loaded_library(lib_name) -> Optional[str]:
+    """
+    According to according to https://man7.org/linux/man-pages/man5/proc_pid_maps.5.html,
+    the file `/proc/self/maps` contains the memory maps of the process, which includes the
+    shared libraries loaded by the process. We can use this file to find the path of the
+    a loaded library.
+    """ # noqa
+    found = False
+    with open("/proc/self/maps") as f:
+        for line in f:
+            if lib_name in line:
+                found = True
+                break
+    if not found:
+        # the library is not loaded in the current process
+        return None
```

Comment:
nit: `found` can be simplified.
```python
        for line in f:
            if lib_name in line:
                break
        else:
            return None
```
Or

```python
if not any([lib_name in line for line in f]):
    return None
```

Reply:
we are finding the first line, so this is necessary.

Reply:
This is not a good practice then because "line" is not supposed to be used out of the loop (although it can be). I'd suggest use "found_line = None" instead

Reply:
sounds good, fixed in [900c257](https://github.com/vllm-project/vllm/pull/11743/commits/900c257640c59a6004564c9edfa5f2945c4618ca)